### PR TITLE
Fix async

### PR DESCRIPTION
**Pull request by HelioGuilherme66**
Hopefully this PR will fix Async tests for Chrome, Phantomjs and IE.

### CODE REVIEW

**Comment by just-be-dev** [comment]
So huge, haha. @emanlove, @ombre42 can I get some help reviewing this issue? @pekkaklarck feel free to pitch in too if you get the chance.

File: test/resources/robotstatuschecker.py [hunk 1]
```diff
@@ -0,0 +1,266 @@
+#!/usr/bin/env python
```
Comment (just-be-dev) [inline test/resources/robotstatuschecker.py]: Why was this file renamed?

File: test/util.sh [hunk 2]
```diff
@@ -0,0 +1,29 @@
+#!/usr/bin/bash
```
Comment (just-be-dev) [inline test/util.sh]: Heh, I wish I would have had this script a week or two ago! I actually wrote my own to correct all these references (see [2456984](https://github.com/rtomac/robotframework-selenium2library/commit/245698483ff52f6e9512a9314ff13c49c29f37a8)). I'm not opposed to adding this to correct any future mistakes, but I don't think it should be done in this PR. Mind moving this script out into its own PR?

Comment (HelioGuilherme66) [inline test/util.sh]: Hi Zephraph,

I am sorry for all this messed PR. I planned (again) to squash it and leave
out the "keyword_press_keys" branch. I failed to have it cleaned up :(.

The main differences in my fork are:
- I use RIDE so, most of the test cases will have 4 spaces and headings
  like "**\* Settings **_" as opposed to "_ Setting *"
- My goal is to have Selenium2Library tests passing OK with all browsers
  (so far tested with Firefox, Chrome, Opera, PhantomJS in Linux, and IE in
  Windows).

Since I started using the .robot extension, I could not have my fork merged
with the original, that is why I was trying to get this solved.

Please do what you feel best, cherry-pick if you can spare that effort or
simply reject the PR.

Sorry for all this work (I am still new to Git ;) ).

Comment (just-be-dev) [inline test/util.sh]: It's all good. 

I'll take some time in the next few days and help you clean it up.

**Comment by HelioGuilherme66** [comment]
I need this merged for the goal of having all tests passing with Chrome, PhantomJS and IE.
Thanks.

**Comment by HelioGuilherme66** [comment]
Please review, @pekkaklarck.

**Comment by pekkaklarck** [comment]
I wouldn't test error messages originating from different browser drivers in this detailed manner. We typically shouldn't need to care about the exact message, testing them in detail means tests break if messages change, and, most importantly, our tests get _very_ complicated. I'd just use something simple like `TimeoutException: *`.

**Comment by HelioGuilherme66** [comment]
@zephraph What you think? Should tests be simplified as @pekkaklarck suggests? I agree, they will remove `if`conditions, and we'll just be expecting the exceptions, it's faster.

**Comment by aaltat** [comment]
I agree with Pekka, `TimeoutException: *`, should be enough.

**Comment by just-be-dev** [comment]
Yes, I agree with the above statements. Don't check the exact error message. If the error message changes slightly we don't want the tests to break.

**Comment by HelioGuilherme66** [comment]
I think this is final version. Thanks.

File: test/acceptance/keywords/async_javascript.robot [hunk 3]
```diff
@@ -61,44 +61,37 @@ Should Be Able To Return Arrays Of Primitives From Async Scripts
...
+    ...    window.location = 'javascript/dynamic';
 
 Should Catch Errors When Executing Initial Script
-    [Documentation]     Should Catch Errors When Executing Initial Script
-    Run Keyword And Expect Error    WebDriverException: Message: you should catch this!*
-    ...    Execute Async Javascript    throw Error('you should catch this!');
+    [Documentation]    Should Catch Errors When Executing Initial Script
+    Run Keyword And Expect Error    WebDriverException:*    Execute Async Javascript
+    ...    throw Error('you should catch this!');
     #TODO Implement Selenium asynchronous javascript test
```
Comment (just-be-dev) [inline test/acceptance/keywords/async_javascript.robot:95]: What are these comments? Is this actually a TODO item? Does it have anything to do with this PR?

**Comment by just-be-dev** [comment]
Aside from those comments on the end of this file (which are probably old holdovers that have nothing to do with this PR) it all looks good to me. 

@pekkaklarck want to confirm and merge?

**Comment by pekkaklarck** [comment]
Looks good to me. There are some unrelated changes to docs but I guess they are OK. Commits should be squashed before the merge, though.

TODOs at the end look pretty weird but they don't seem to have anything to do with this PR.

**Comment by HelioGuilherme66** [comment]
Squashed and ready to merge. Thanks.

**Comment by pekkaklarck** [comment]
There has been some problem in squashing because this PR still has 5 commits. You can see that in the tab header on the top of this page. Most of the commits aren't actually related to this PR. Could it be that you have forgot to merge with master or merge has been done a little wrong? Regardless the cause, this PR is now in worse state than it was earlier and I don't think it can be merged. I don't know could the situation be fixed somehow or would it be better to create a new PR with just the relevant changes.

I understand that squashing, merging and other a bit more advanced Git topics are complicated and it is easy to make mistakes. I'm not a Git guru myself and often make such mistakes too.

**Comment by aaltat** [comment]
I also make the same mistake many times and for me,  usually the easiest way is create new branch from master and `git cherry-pick` the commit to the new branch and create a new pull request.

Because git is powerfull, I would assume that there are many other ways to solve the problem.

**Comment by HelioGuilherme66** [comment]
I am going to force push of previous version, and later will squash and rebase.

**Comment by HelioGuilherme66** [comment]
Finally it is ready to merge!
